query-parser/src/parse.rs

```diff
@@ -1073,4 +1157,79 @@ mod test {
                               }));
 
     }
+
+    #[test]
+    fn test_pull() {
+        assert_edn_parses_to!(Query::pull_attribute,
+                              "*",
+                              PullAttributeSpec::Wildcard);
+        assert_edn_parses_to!(Query::pull_attributes,
+                              "[*]",
+                              vec![PullAttributeSpec::Wildcard]);
+        assert_edn_parses_to!(Find::elem,
+                              "(pull ?v [*])",
+                              Element::Pull(Pull {
+                                  var: Variable::from_valid_name("?v"),
+                                  patterns: vec![PullAttributeSpec::Wildcard],
+                              }));
+
+        let foo_bar = ::std::rc::Rc::new(edn::NamespacedKeyword::new("foo", "bar"));
+        let foo_baz = ::std::rc::Rc::new(edn::NamespacedKeyword::new("foo", "baz"));
+        assert_edn_parses_to!(Query::pull_concrete_attribute,
+                              ":foo/bar",
+                              PullAttributeSpec::Attribute(
+                                  PullConcreteAttribute::Ident(foo_bar.clone())));
+        assert_edn_parses_to!(Query::pull_attribute,
+                              ":foo/bar",
+                              PullAttributeSpec::Attribute(
+                                  PullConcreteAttribute::Ident(foo_bar.clone())));
+        assert_edn_parses_to!(Find::elem,
```

Comment:
Consider asserting failure in the mixed wildcard and attribute case: `(pull ?v [:foo/bar *])`.

Reply:
Done.